File: fixtures/scale/csharp/small/Util/TextMeasure.cs
namespace TextPrint.Util
{
    public static class TextMeasure
    {
        public static int VisibleLength(string line)
        {
            var length = 0;
            var inEscape = false;
            foreach (var ch in line)
            {
                if (ch == '\u001b')
                {
                    inEscape = true;
                }
                else if (inEscape && ch == 'm')
                {
                    inEscape = false;
                }
                else if (!inEscape)
                {
                    length++;
                }
            }
            return length;
        }
    }
}

File: fixtures/scale/swift/small/Util/TextMeasure.swift
import Foundation

enum TextMeasure {
    static func visibleLength(_ line: String) -> Int {
        var length = 0
        var inEscape = false
        for ch in line {
            if ch == "\u{1b}" {
                inEscape = true
            } else if inEscape && ch == "m" {
                inEscape = false
            } else if !inEscape {
                length += 1
            }
        }
        return length
    }
}
